src/elements.c

```diff
@@ -235,6 +295,40 @@ int wally_asset_rangeproof(uint64_t value,
     return ret;
 }
 
+int wally_explicit_rangeproof(uint64_t value,
+                              const unsigned char *nonce, size_t nonce_len,
+                              const unsigned char *vbf, size_t vbf_len,
+                              const unsigned char *commitment, size_t commitment_len,
+                              const unsigned char *generator, size_t generator_len,
+                              unsigned char *bytes_out, size_t len, size_t *written)
+{
+    return wally_asset_rangeproof_with_nonce(value, nonce, nonce_len, NULL, 0, NULL, 0,
+                                             vbf, vbf_len, commitment, commitment_len, NULL, 0,
+                                             generator, generator_len, value, -1, 0,
+                                             bytes_out, len, written);
+}
+
+int wally_explicit_rangeproof_verify(const unsigned char *rangeproof, size_t rangeproof_len,
+                                     uint64_t value,
+                                     const unsigned char *commitment, size_t commitment_len,
+                                     const unsigned char *generator, size_t generator_len)
+{
+    const secp256k1_context *ctx = secp_ctx();
+    secp256k1_pedersen_commitment commit;
+    secp256k1_generator gen;
+    uint64_t min_v, max_v;
+
+    if (!rangeproof || rangeproof_len > ASSET_EXPLICIT_RANGEPROOF_MAX_LEN ||
+        parse_commitment(ctx, commitment, commitment_len, &commit) != WALLY_OK ||
+        parse_generator(ctx, generator, generator_len, &gen) != WALLY_OK)
+        return WALLY_EINVAL;
+
+    if (!secp256k1_rangeproof_verify(ctx, &min_v, &max_v, &commit,
+                                     rangeproof, rangeproof_len, NULL, 0, &gen))
+        return WALLY_EINVAL;
+    return value == min_v ? WALLY_OK : WALLY_EINVAL;
```

Comment:
I think we should check `value == max_v` too

Reply:
This follows the elements `blindpsbt` logic which only checks min value.